### ISSUE TITLE
Can I use Header - Authorization for authorization token?

### ISSUE THREAD

**Issue by savager** [ghaiklor/generator-sails-rest-api#224]
I'm not sure how passport is handling it, but I can only get it working passing auth_token as a parameter at the moment. If this isn't supported that's fine just let me know.

**Comment by ghaiklor** [comment]
@savager you can, `Authorization: Bearer <token>`

**Comment by savager** [comment]
That's what I thought, but I get, "no auth token" still.. don't worry i'll figure it out. Thanks!

**Comment by ghaiklor** [comment]
@savager strange, if you find some useful, let me know, please.

**Comment by savager** [comment]
Might have been a change in the new version, but I have it fixed by using a different method in passport.js.

`const JWT_STRATEGY_CONFIG = {
  secretOrKey: '9dc0f33dd66707eda39e7ecb6746818299999094bbcca24657f6969469a9870d',
  jwtFromRequest: ExtractJwt.versionOneCompatibility({tokenBodyField: 'access_token'}),
  tokenQueryParameterName: 'access_token',
  jwtFromRequest: ExtractJwt.fromAuthHeaderWithScheme('Bearer'),
  session: false,
  passReqToCallback: true
};`

**Comment by ghaiklor** [comment]
@savager did you figure out how to fix the problem ?

**Comment by savager** [comment]
If i remember correctly I changed the config/passport.js JWT extract to use fromAuthHeader() instead of fromAuthHeaderWithScheme(authScheme)

this way worked as Authorization JWT :token:

**Comment by tlays11** [comment]
In config/passport.js the  JWT_STRATEGY_CONFIG  const, updated the jwtFromRequest property to this:
 jwtFromRequest: ExtractJwt.versionOneCompatibility({authScheme : 'Bearer', tokenBodyField: 'api_key'}),


This way you can still used api_key get param.

**Comment by ghaiklor** [comment]
@tlays11 can you create a PR with that change? Does it break `access_token` field ?

**Comment by mikedevita** [comment]
for me changing to the `fromAuthHeader()` didn't work i had to do like @tlays11 suggested and do the following

**in config/passport.js**  change
```
const JWT_STRATEGY_CONFIG = {
  secretOrKey: 'c4a071df2a2ef9d0dcce2e38d39d67cee6bba0ddcbe69f21cb2e20ea3d9cd9ef',
  jwtFromRequest: ExtractJwt.versionOneCompatibility({authScheme: 'Bearer', tokenBodyField: 'access_token'}),
  tokenQueryParameterName: 'access_token',
  session: false,
  passReqToCallback: true
};
```

to 
```
const JWT_STRATEGY_CONFIG = {
  secretOrKey: '2546dd38b356dafb9ea2f6ed75586f883f3c3f1a55f1edc630b3d7eb3689d54e',
  jwtFromRequest: ExtractJwt.versionOneCompatibility({
    authScheme: 'Bearer',
    tokenBodyField: 'access_token'
  }),
  tokenQueryParameterName: 'access_token',
  authScheme: 'Bearer',
  session: false,
  passReqToCallback: true
};
```

**Comment by ghaiklor** [comment]
@mikedevita great, yeah, looks good. You can create a PR for this. I'm not sure if we need to ask a user for this, imho, it should work for both cases without prompting the user.

**Comment by mikedevita** [comment]
@ghaiklor yeah i dont think its required if the current ver supports both. 👍
